k4GeneratorsConfig/src/pythiaLHERunner.cxx

```diff
@@ -1,89 +1,110 @@
-// main07.cc is a part of the PYTHIA event generator.
-// Copyright (C) 2024 Torbjorn Sjostrand.
-// PYTHIA is licenced under the GNU GPL v2 or later, see COPYING for details.
-// Please respect the MCnet Guidelines, see GUIDELINES for details.
-
-// Keywords: two-body decay; astroparticle; python; matplotlib
-
+// Pythia8
 #include "Pythia8/Pythia.h"
 #include "Pythia8Plugins/HepMC3.h"
-#include <unistd.h>
 
-using namespace Pythia8;
+// std
+#include <string>
 
-//==========================================================================
+// *nix
+#include <unistd.h>
 
 int main(int argc, char** argv) {
+  // Default values for the command-line arguments
+  std::string pythiaCmdFilePath = "Pythia.dat";
+  std::string lheFilePath = "Pythia.lhe";
+  std::string hepmcFilePath = "Pythia.hepmc";
+  bool verbose = false;
 
-  // read the options
-  std::string filename = "Pythia.dat";
-  std::string lhefilename = "Pythia.lhe";
-  std::string hepmcFile = "Pythia.hepmc";
-  int c;
-  while ((c = getopt(argc, argv, "f:l:o:")) != -1)
-    switch (c) {
+  // Read the command-line arguments
+  int opt;
+  while ((opt = getopt(argc, argv, "f:l:o:vh")) != -1) {
+    switch (opt) {
     case 'f':
-      filename = optarg;
+      pythiaCmdFilePath = std::string(optarg);
       break;
     case 'l':
-      lhefilename = optarg;
+      lheFilePath = std::string(optarg);
       break;
     case 'o':
-      hepmcFile = optarg;
+      hepmcFilePath = std::string(optarg);
+      break;
+    case 'v':
+      verbose = true;
       break;
     case 'h':
-      std::cout << "Usage: pythiaLHERunner -h -f filename -l filename" << std::endl;
-      std::cout << "-h: print this help" << std::endl;
-      std::cout << "-f filename: input file containing the pythia commands" << std::endl;
-      std::cout << "-l filename: input file containing the LHE events" << std::endl;
-      std::cout << "-o filename: output file containing the hepmc events" << std::endl;
-      exit(0);
     default:
+      std::cout << "Usage: pythiaLHERunner [-h, -v] " << "-f FILEPATH -l FILEPATH [-o FILEPATH]\n"
+                << "  -h: print this help and exit\n"
+                << "  -v: more verbose output\n"
+                << "  -f FILEPATH: input file containing the Pythia " << "commands\n"
+                << "  -l FILEPATH: input file containing the LHE events\n"
+                << "  -o FILEPATH: output file containing the HepMC events" << std::endl;
+
```

Comment:
The default case also captures wrong arguments and should exit with non-zero as that is most likely an error. However, the help should exit with 0 as `-h` is almost certainly a deliberate action.

I usually have a simple usage and help message literal string that can be printed here, e.g. [as in `podio-dump-tool`](https://github.com/AIDASoft/podio/blob/dd99663c533785887bcb2c9e8fa7370f4249bd43/tools/src/podio-dump-tool.cpp#L33-L50)